### PR TITLE
display the next shipping time range instead of the opening hours

### PR DESCRIPTION
**Pull request by vladimir-8**
| |with pre-ordering |without pre-ordering |
|-|-|-|
|Closed; regular opening hours |Simulator Screenshot - iPhone 16 - 2024-11-08 at 14 23 55 |Simulator Screenshot - iPhone 16 - 2024-11-08 at 14 23 39 |
|Closed; closing rule/'holiday" |Simulator Screenshot - iPhone 16 - 2024-11-08 at 14 24 05 | |

### CODE REVIEW

**Review by Atala — COMMENTED** [review]
hello,

i am confused the screenshot in the PR show the code working to detect "holidays" but it does not seem to work in this case in the code? for example I didn't find the string for "closed on holiday" from your screenshot. 

also the comment
```
    //FIXME: this is based on the regular opening hours
    // and does not take closing rules ("Holidays") into account
```

to me it is normal as `OpeningHoursSpecification` is like the "usual opening hours" whereas `getNextShippingDate` is the available time with all filters applied (holidays, rush mode, whatever) ?

as a side note maybe add a comment on top of `OpeningHoursSpecification` class, i actually didnt really understand what this class is about and for what are all these complicated methods (translate API responses to human readable format?)

**Comment by vladimir-8** [comment]
> I didn't find the string for "closed on holiday" from your screenshot.

It's a restaurant name 😄

**Comment by vladimir-8** [comment]
This is how all these cases work under-the-hood:

| |with pre-ordering |without pre-ordering |
|-|-|-|
|Closed; regular opening hours |uses `getNextShippingTime` (time ranges returned by the backend) |uses `OpeningHoursSpecification` |
|Closed; closing rule/'holiday" |uses `getNextShippingTime` (time ranges returned by the backend; backend takes care of the closing rule/'holiday") |uses `OpeningHoursSpecification`; ❗the date/time would be wrong at it does not take closing rule/'holiday" into account; that what this FIXME is about: https://github.com/coopcycle/coopcycle-app/pull/1915/files#diff-0b5ed3e1cf95e0b895bbb90537deea76c130777048cdd7bdcc0c6b151f96b63fR105-R106 We need to fix it at some point, but it's not that critical, I think, as the customer, at least, can not make a wrong order. I see there is a `specialOpeningHoursSpecification` method, that only used by the Restaurant app, that's where we should probably implement it |

**Comment by vladimir-8** [comment]
> i actually didnt really understand what this class is about and for what are all these complicated methods (translate API responses to human readable format?)

Yes, and a few helper functions. I did not look into it too much, but from my tests, it seems that it does its job

File: src/utils/checkout.js [hunk 1]
```diff
@@ -43,18 +36,38 @@ function timingAsText(timing, now) {
...
+
+  const timing = restaurant.timing.delivery || restaurant.timing.collection;
+
+  if (!timing) {
+    return null;
+  }
+
+  // FIXME
+  // This hotfixes a bug on the API
+  // https://github.com/coopcycle/coopcycle-web/issues/2213
```
[the remark this answers is not in the document]
Comment (vladimir-8) [reply src/utils/checkout.js:49]: Well, I neither fixed nor tested it. I just moved the code to another method

**Review by Atala — APPROVED** [review]

**Pull request merged**